### PR TITLE
Fix LoginActivity fragments not restoring properly

### PR DESCRIPTION
**Pull request by phillipm**
__Contains commits from https://github.com/dimagi/commcare-odk/pull/993__

Turns out when you don't call `super.onSaveInstanceState` when overriding `onSaveInstanceState`, then none of your fragments get restored :boom:

The result is that rotating the screen while any login task is running detaches the task from the activity, making the progress dialog disappear

Fix for http://manage.dimagi.com/default.asp?200782

### CODE REVIEW

File: app/src/org/commcare/dalvik/activities/LoginActivity.java [hunk 1]
```diff
@@ -132,7 +128,9 @@ public void onRequestPermissionsResult(int requestCode,
     }
 
     @Override
-    public void onSaveInstanceState(Bundle savedInstanceState) {
+    protected void onSaveInstanceState(Bundle savedInstanceState) {
+        super.onSaveInstanceState(savedInstanceState);
```
Comment (phillipm) [inline app/src/org/commcare/dalvik/activities/LoginActivity.java]: The fix

Comment (amstone326) [inline app/src/org/commcare/dalvik/activities/LoginActivity.java]: Yay! Confused by how this was causing it though, can you explain?

Comment (phillipm) [inline app/src/org/commcare/dalvik/activities/LoginActivity.java]: I'm guessing that in the Android platform's implementation of `onSaveInstanceState` certain fragment info is committed to the bundle. By overriding `onSaveInstanceState` but not passing the bundle to the `super` implementation, we interrupted that code path, preventing the platform from storing whatever data it usually stores into that bundle.

**Comment by phillipm** [comment]
Smaller version https://github.com/dimagi/commcare-odk/pull/999

**Pull request closed without merging**